### PR TITLE
manifest: nrf_modem: v1.4.0

### PR DESCRIPTION
**Pull request by lemrey**
Update manifest for libmodem v1.4.0 release.

~~_commit a2d0fef4737 to be removed after https://github.com/nrfconnect/sdk-nrf/pull/5983 is merged_~~

### CODE REVIEW

**Review by balaji-nordic — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

File: lib/nrf_modem_lib/Kconfig [hunk 1]
```diff
@@ -69,6 +69,12 @@ endif # NRF_MODEM_LIB_TRACE_MEDIUM_RTT
 
 endif # NRF_MODEM_LIB_TRACE_ENABLED
 
+config NRF_MODEM_LIB_AT_CMD_FILTER
+	bool "Enable AT command filters"
+	help
+	  When  enabled, filtered AT commands are redirected to an application-provided
```
Comment (rlubos) [inline lib/nrf_modem_lib/Kconfig:75]:
```suggestion
	  When enabled, filtered AT commands are redirected to an application-provided
```

File: lib/nrf_modem_lib/nrf_modem_at_cmd_filter.c [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+			/* Store size. */
+			sms_buffer_list[i].pdu_size =
+					strtol(&(at_cmd[8]), NULL, 10);
+			/* Store in SMS buffer. */
+			strncpy(sms_buffer_list[i].data, s1,
+				(size_t) (s2 - s1));
+			/* Make AT response. */
+			return response_buffer_fill(buf, len,
+					"+CMGW: %d\r\nOK\r\n", i+1);
+			return 0;
```
Comment (rlubos) [inline lib/nrf_modem_lib/nrf_modem_at_cmd_filter.c:268]: The return was just above.

File: subsys/mgmt/fmfu/src/fmfu_mgmt.c [hunk 3]
```diff
@@ -108,21 +108,6 @@ static int unpack(struct mgmt_ctxt *ctxt, struct firmware_packet *packet,
 	return offset + packet->data_len;
 }
 
-static int get_mgmt_err_from_modem_ret_err(int err)
```
Comment (hakonfam) [inline subsys/mgmt/fmfu/src/fmfu_mgmt.c:111]: Does `nrf_modem_full_dfu_bl_write` no longer return an error message of any use? With this proposal what ever is returned from this function is "overwritten" with `MGMT_ERR_EBADSTATE`

File: subsys/mgmt/fmfu/src/fmfu_mgmt.c [hunk 4]
```diff
@@ -174,10 +159,9 @@ static int fmfu_firmware_upload(struct mgmt_ctxt *ctx)
 							 packet.data_len,
 							 packet.data);
 			if (rc != 0) {
-				LOG_ERR("Error in writing data, rc: %d,"
-					" errno: %d", rc, errno);
-				return get_mgmt_err_from_modem_ret_err(
-						rc);
+				LOG_ERR("Error in writing data,"
+					" err: %d", rc);
```
Comment (hakonfam) [inline subsys/mgmt/fmfu/src/fmfu_mgmt.c:163]:
```suggestion
				LOG_ERR("Error in writing data, err: %d", rc);
```

File: lib/nrf_modem_lib/nrf_modem_at_cmd_filter.c [hunk 5]
```diff
@@ -0,0 +1,305 @@
+/*
```
Comment (rlubos) [inline lib/nrf_modem_lib/nrf_modem_at_cmd_filter.c:1]: The commit message dose not mention `nrf_modem_at_cmd_filter` introduction at all (and there is no entry in the changelog). Anyway, it's seems pretty separate from manifest update, would it make sense to make a separate commit out of it?

**Review by hakonfam — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]

File: lib/nrf_modem_lib/Kconfig [hunk 6]
```diff
@@ -69,6 +69,12 @@ endif # NRF_MODEM_LIB_TRACE_MEDIUM_RTT
 
 endif # NRF_MODEM_LIB_TRACE_ENABLED
 
+config NRF_MODEM_LIB_AT_CMD_FILTER
+	bool "Enable AT command filters"
```
Comment (tejlmand) [inline lib/nrf_modem_lib/Kconfig:73]: As per the guidelines, please remove `Enable`:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/build/kconfig/tips.html#prompt-strings

> For a Kconfig symbol that enables a driver/subsystem FOO, consider having just “Foo” as the prompt, instead of “Enable Foo support” or the like.

File: tests/lib/nrf_modem_lib/at_cmd_filter/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(nrf_modem_lib)
+
+FILE(GLOB app_sources src/*.c)
+target_sources(app PRIVATE ${app_sources})
+
+target_include_directories(app
+  PRIVATE
+  ${ZEPHYR_BASE}/../nrf/include/modem/
+)
```
Comment (tejlmand) [inline tests/lib/nrf_modem_lib/at_cmd_filter/CMakeLists.txt:18]: Why the new lines lines here ?
Your line doesn't seem to be that long.

Also use: `${ZEPHYR_NRF_MODULE_DIR}/include/modem/` instead.

**Review by tejlmand — COMMENTED** [review]
some minor CMake / Kconfig observations

**Comment by lemrey** [comment]
Rebased, split the new `at_sms_cert` out in a separate commit.

**Review by nordic-hani — APPROVED** [review]

**Comment by lemrey** [comment]
Blocked by: https://github.com/nrfconnect/sdk-nrf/pull/6168

**Review by umapraseeda — APPROVED** [review]

**Comment by richardmccrae-nordicsemi** [comment]
libmodem system ztest test_nrf_modem_delta_dfu_update must be skipped.  It will fail on most recent customer MFW release
See: 
https://projecttools.nordicsemi.no/jira/browse/TNSW-39814?filter=-3

**Comment by richardmccrae-nordicsemi** [comment]
Appears that there is still a merge conflict with west.yml.

**Review by richardmccrae-nordicsemi — CHANGES_REQUESTED** [review]
Merge conflict with west.yml

**Review by jhirsi — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by lemrey** [comment]
PIcked the commits into #6156, closing this.
